src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,117 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader';
+import { ErrorsMessages } from '../../types/ErrorsMessages';
+
+type Props = {
+  todo: Todo,
+  handleChecker: (id: number, data: unknown) => void,
+  removeTodo: (id: number) => void,
+  processings: number[],
+  errorMessage: (message: ErrorsMessages) => void
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleChecker,
+  removeTodo,
+  processings,
+  errorMessage,
+}) => {
+  const { id, completed, title } = todo;
+  const [isFormActive, setIsFormActive] = useState(false);
+  const [inputValue, setInputValue] = useState(title);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    setInputValue(event.target.value);
+  };
+
+  const validateTitle = () => {
+    if (!inputValue.trim().length) {
+      removeTodo(id);
+      errorMessage(ErrorsMessages.Title);
+
+      return false;
+    }
+
+    if (title === inputValue.trim()) {
+      setIsFormActive(false);
+
+      return false;
+    }
+
+    setIsFormActive(false);
+
+    return true;
+  };
+
+  const changeTodoTitle = () => {
+    return validateTitle() ? (
+      handleChecker(id, { title: inputValue })
+    ) : (
+      null
+    );
+  };
+
+  const moveCursorToEndOfInput = (e: React.FocusEvent<HTMLInputElement>) => (
+    e.currentTarget.setSelectionRange(
+      e.currentTarget.value.length,
+      e.currentTarget.value.length,
+    )
+  );
+
+  return (
+    <div className={classNames('todo', { completed })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={() => handleChecker(id, { completed: !completed })}
+        />
+      </label>
+
+      {isFormActive ? (
+        <form onSubmit={(event) => {
+          event.preventDefault();
+          changeTodoTitle();
+        }}
+        >
```

Comment:
it would be better to create separate function
```suggestion
        <form onSubmit={onSubmit}>
```